src/server/req_shutdown.c

```diff
@@ -323,9 +323,9 @@ shutdown_chkpt(job *pjob)
 		if (pjob->ji_qs.ji_state == JOB_STATE_TRANSIT)
 			svr_setjobstate(pjob, JOB_STATE_RUNNING, JOB_SUBSTATE_RUNNING);
 
-		pjob->ji_qs.ji_substate = JOB_SUBSTATE_RERUN;
 		pjob->ji_qs.ji_svrflags |= JOB_SVFLG_HASRUN;
 		pjob->ji_qs.ji_svrflags |= JOB_SVFLG_CHKPT;
+		pjob->ji_qs.ji_svrflags |= JOB_SVFLG_HASHOLD;
```

Comment:
I think it would be best to OR all these flags together and then set ji_svrflags.